database/factories/PostFactory.php

```diff
@@ -2,18 +2,27 @@
 
 /** @var \Illuminate\Database\Eloquent\Factory $factory */
 
+use App\Models\Comment;
 use App\Models\Post;
 use App\Models\Status;
 use App\Models\User;
 use Faker\Generator as Faker;
+use Illuminate\Support\Carbon;
 
 $factory->define(Post::class, function (Faker $faker) {
-    $status = $faker->randomElement([Status::DRAFT, Status::PUBLISHED]);
     return [
-        'name' => $faker->name,
-        'content' => $faker->text,
-        'user_id' => User::get()->first()->id, // factory
-        'published_at' => $status === Status::PUBLISHED ? $faker->dateTimeBetween('-5 months', 'now') : null,
-        'status_id' => $status,
+        'name'         => $faker->name,
+        'content'      => $faker->text,
+        'user_id'      => function () {
+            return factory(User::class)->create()->id;
+        },
+        'published_at' => Carbon::now()->subMonths($faker->randomDigit),
+        'status_id'    => Status::PUBLISHED,
     ];
 });
+
+$factory->afterCreating(Post::class, function ($post, $faker) {
```

Comment:
it may be helpful to make a state called `with-comments` or something then you can use that to trigger the creation of comments for the post. Just in case you have scenarios you want to test where a post doesn't have any comments. It also may speed things up since you wouldn't be creating comments every time you factory a post.

Reply:
That's true, and it seems like a better solution to what I'm trying to do. Thou when I looked up factory states the docs don't say anything about passing the newly-created posts. Is there something that I'm not aware of?
https://laravel.com/docs/7.x/database-testing#factory-states